### PR TITLE
Fixes #1215.  Allow secret key to come from file.

### LINKED ISSUE #1215: Secrets file should be configurable.
For docker config and other secrets systems, the secrets come into a file inside the docker container.  

This would be nice to be configured.  Right now it has to be copied, but it could just be done in the following manner.

NOTE:  Whether you do it based upon the config.yaml or the envvars, is up to you, but if you want to scale it out past 1 docker container, this will make it easier for people to deploy it in docker.

``` python
if os.getenv("INVENTREE_SECRET_KEY",""):
    # The secret key is passed in as an envvar directly
    SECRET_KEY = os.getenv("INVENTREE_SECRET_KEY")
elif os.getenv("INVENTREE_SECRET_KEY_FILE",""):
    # Docker compose secrets come in as a file to the container
    SECRET_KEY = open(os.getenv("INVENTREE_SECRET_KEY_FILE", "r")).read().strip()
else:
    # Works only for one docker instance which is probably okay.  
    SECRET_KEY = generate_secret_key()
```

The docker compose config would then look like this:

``` docker-compose
inventree:
    build: ./inventree-docker
    container_name: inventree
    expose:
      - 9767/tcp
    networks:
      - proxy
      - db
    depends_on:
      - db
    volumes:
      - /mnt/dockerdata/inventree/static:/usr/src/static
      - /mnt/dockerdata/inventree/media:/usr/src/media
    secrets:
      - postgres_password
      - postgres_user
      - django_secret_key
    environment:
      - SECRET_KEY_FILE=/run/secrets/django_secret_key
      - DATABASE_USER_FILE=/run/secrets/postgres_user
      - DATABASE_PASSWORD_FILE=/run/secrets/postgres_password
      [...]
```

### PR DESCRIPTION
**Pull request by rcludwick**
Fixes #1215

### CODE REVIEW

File: InvenTree/InvenTree/settings.py [hunk 1]
```diff
@@ -59,20 +57,31 @@
...
 # Get a logger instance for this setup file
 logger = logging.getLogger(__name__)
 
-# Read the autogenerated key-file
-key_file_name = os.path.join(BASE_DIR, 'secret_key.txt')
-logger.info(f'Loading SECRET_KEY from {key_file_name}')
-key_file = open(key_file_name, 'r')
+# Default action is to run the system in Debug mode
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = os.getenv("INVENTREE_DEBUG") if os.getenv("INVENTREE_DEBUG") else CONFIG.get("debug", True)
```
[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py:65]: Yeah, now that I'm looking at that, I don't like it either.  I'll fix it tomorrow.

File: InvenTree/InvenTree/settings.py [hunk 2]
```diff
@@ -59,20 +57,31 @@
...
-SECRET_KEY = key_file.read().strip()
+if os.getenv("INVENTREE_SECRET_KEY"):
+    # Secret key passed in directly
+    SECRET_KEY = os.getenv("INVENTREE_SECRET_KEY").strip()
+    logger.info("SECRET_KEY loaded by INVENTREE_SECRET_KEY")
+else:
+    # Secret key passed in by file location
+    key_file = os.getenv("INVENTREE_SECRET_KEY_FILE")
+    if key_file and os.path.isfile(key_file):
+        logger.info("SECRET_KEY loaded by INVENTREE_SECRET_KEY_FILE")
```
[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py:75]: That's simple enough to do.

File: InvenTree/InvenTree/settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py:65]: Fixed in f89948e

File: InvenTree/InvenTree/settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py:75]: Fixed in f89948e

File: InvenTree/InvenTree/settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py:65]: Fixed in f89948e

[the remark this answers is not in the document]
Comment (rcludwick) [reply InvenTree/InvenTree/settings.py]: Fixed in https://github.com/inventree/InvenTree/commit/5d1be3e244f55d273750860f2a9f72d157c9ad1d

**Comment by rcludwick** [comment]
Also the issue where the secret key file couldn't be loaded is added in https://github.com/inventree/InvenTree/commit/5d1be3e244f55d273750860f2a9f72d157c9ad1d as well.

**Comment by SchrodingersGat** [comment]
Thanks for the contribution @rcludwick

**Pull request merged**